File: app/src/main/java/com/jmalinen/blackjack/model/GameState.kt
package com.jmalinen.blackjack.model

data class ExtraPlayerState(
    val hand: Hand = Hand(),
    val result: HandResult? = null
)

data class GameState(
    val phase: GamePhase = GamePhase.BETTING,
    val rules: CasinoRules = CasinoRules(),
    val playerHands: List<Hand> = emptyList(),
    val activeHandIndex: Int = 0,
    val dealerHand: Hand = Hand(),
    val chips: Int = 1000,
    val currentBet: Int = 10,
    val insuranceBet: Int = 0,
    val availableActions: Set<PlayerAction> = emptySet(),
    val handResults: Map<Int, HandResult> = emptyMap(),
    val roundMessage: String = "",
    val showDealerHoleCard: Boolean = false,
    val roundPayout: Int = 0,
    val handsPlayed: Int = 0,
    val handsWon: Int = 0,
    val coachEnabled: Boolean = false,
    val coachFeedback: String = "",
    val coachCorrect: Int = 0,
    val coachTotal: Int = 0,
    val deviationsEnabled: Boolean = false,
    val shoePenetration: Float = 0f,
    val runningCount: Int = 0,
    val trueCount: Float = 0f,
    val showCount: Boolean = false,
    val extraPlayers: List<ExtraPlayerState> = emptyList()
) {
    val activeHand: Hand? get() = playerHands.getOrNull(activeHandIndex)
    val totalBetOnTable: Int get() = playerHands.sumOf { it.bet } + insuranceBet
}

File: ios/Blackjack/Blackjack/Model/GameState.swift
struct ExtraPlayerState: Equatable {
    var hand: Hand = Hand()
    var result: HandResult? = nil
}

struct GameState: Equatable {
    var phase: GamePhase = .betting
    var rules: CasinoRules = CasinoRules()
    var playerHands: [Hand] = []
    var activeHandIndex: Int = 0
    var dealerHand: Hand = Hand()
    var chips: Int = 1000
    var currentBet: Int = 10
    var insuranceBet: Int = 0
    var availableActions: Set<PlayerAction> = []
    var handResults: [Int: HandResult] = [:]
    var roundMessage: String = ""
    var showDealerHoleCard: Bool = false
    var roundPayout: Int = 0
    var handsPlayed: Int = 0
    var handsWon: Int = 0
    var coachEnabled: Bool = false
    var coachFeedback: String = ""
    var coachCorrect: Int = 0
    var coachTotal: Int = 0
    var deviationsEnabled: Bool = false
    var shoePenetration: Float = 0
    var runningCount: Int = 0
    var trueCount: Float = 0
    var showCount: Bool = false
    var extraPlayers: [ExtraPlayerState] = []

    var activeHand: Hand? {
        activeHandIndex < playerHands.count ? playerHands[activeHandIndex] : nil
    }

    var totalBetOnTable: Int {
        playerHands.reduce(0) { $0 + $1.bet } + insuranceBet
    }
}
